pkg/worker/request.go

```diff
@@ -55,8 +56,11 @@ func DoRequest(url string, client *http.Client, callback func() (io.Reader, stri
 
 		// And if we can't even do that, log it.
 		resp, err := pesterClient.Do(req)
-		if err != nil || resp.StatusCode != http.StatusOK {
-			errlog.LogError(errors.Wrapf(err, "could not send error message to master URL (%v)", url))
+		if err == nil && resp.StatusCode != http.StatusOK {
+			err = fmt.Errorf("unexpected status code %d", resp.StatusCode)
```

Comment:
if err is nil, errors.Wrap() returns nil. Passing nil to errlog.LogError() causes a segfault.

that was a fun bug to track down 😓

Reply:
wow what

oh well, in that case, i'd return at this point to avoid the import anyway, but if you want to merge that's fine too. up to you, thanks for the fix.

Reply:
of fix the nil handling in the error log, that would be ok too

Reply:
I'm fine with it...